gdpr/tests/test_gdpr_views.py

```diff
@@ -0,0 +1,192 @@
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+@pytest.mark.django_db
+def test_api_delete_user_data(
+    settings,
+    user_factory,  # users
+    area_search_factory,  # plotsearch
+):
+    settings.GDPR_API_DELETE_SCOPE = "gdprdelete"
+    settings.OIDC_API_TOKEN_AUTH = {
+        "API_AUTHORIZATION_FIELD": "authorization.permissions.scopes",
+    }
+
+    user = user_factory(first_name="Kuningas", last_name="Litmanen")
+    user_id = user.id
+    areasearch = area_search_factory(user=user, description_area="Test")
+    apirequest_factory = APIRequestFactory()
+    request = apirequest_factory.delete(f"/v1/pub/gdpr-api/v1/profiles/{user.uuid}")
+    user_authorization = UserAuthorization(
+        user=user,
+        api_token_payload={
+            "amr": ["suomi_fi"],
+            "authorization": {"permissions": {"scopes": ["gdprdelete"]}},
+        },
+    )
+    force_authenticate(request, user=user, token=user_authorization)
+
+    response = MvjGDPRAPIView.as_view()(request, uuid=user.uuid)
+
+    # Expect deletion request to be successful.
+    assert response.status_code == status.HTTP_204_NO_CONTENT
+
+    # Ensure that auditlog entry has been created.
+    assert (
+        LogEntry.objects.get_for_model(user._meta.model)
+        .filter(object_id=user_id, changes="GDPR API: User data deleted.")
+        .exists()
+        is True
+    )
+
+    # Check that user instance is deleted.
+    with pytest.raises(user.DoesNotExist):
+        user.refresh_from_db()
+
+    areasearch.refresh_from_db()
+    assert areasearch.user is None
+
+
+@pytest.mark.django_db
+def test_api_delete_user_data_not_possible(
+    settings,
+    user_factory,  # users
+    information_check_factory,  # plotsearch
+):
+    settings.GDPR_API_DELETE_SCOPE = "gdprdelete"
+    settings.OIDC_API_TOKEN_AUTH = {
+        "API_AUTHORIZATION_FIELD": "authorization.permissions.scopes2",
+    }
+
+    user = user_factory()
+    # IMPORTANT: This model is not expected to be related to a user that would be doing
+    # a GDPR API delete. It is used simply for demonstration purposes, as the model
+    # has `on_delete=models.PROTECT` for the `preparer` field that refers to a `User` instance.
+    information_check = information_check_factory(
+        preparer=user, name=InformationCheckName.CREDITWORTHINESS
+    )
+    apirequest_factory = APIRequestFactory()
+    request = apirequest_factory.delete(f"/v1/pub/gdpr-api/v1/profiles/{user.uuid}")
+    user_authorization = UserAuthorization(
+        user=user,
+        api_token_payload={
+            "amr": ["suomi_fi"],
+            "authorization": {"permissions": {"scopes": ["gdprdelete"]}},
+        },
+    )
+    force_authenticate(request, user=user, token=user_authorization)
+
+    response = MvjGDPRAPIView.as_view()(request, uuid=user.uuid)
+
+    # Expect the test to fail due to on_delete=models.PROTECT on InformationCheck.preparer
```

Comment:
Expect test to fail --> "Expect request to fail"? Because I don't think the test will fail the assert here.